packages/shared/lib/services/sync/data/records.service.integration.test.ts

```diff
@@ -1,56 +1,59 @@
 import { expect, describe, it, beforeAll } from 'vitest';
 import dayjs from 'dayjs';
 import { multipleMigrations } from '../../../db/database.js';
-import * as DataService from './data.service.js';
 import * as RecordsService from './records.service.js';
 import { createConfigSeeds } from '../../../db/seeders/config.seeder.js';
-import type { GetRecordsResponse, DataRecord } from '../../../models/Sync.js';
-import type { ServiceResponse } from '../../../models/Generic.js';
-import connectionService from '../../connection.service.js';
-import { generateInsertableJson, createRecords } from './mocks.js';
+import { upsertRecords } from './mocks.js';
 
 const environmentName = 'records-service';
 
-describe('Records service integration test', () => {
+describe('Records service', () => {
     beforeAll(async () => {
         await multipleMigrations();
         await createConfigSeeds(environmentName);
     });
 
+    it('Should retrieve records', async () => {
+        const n = 10;
+        const { connection, model } = await upsertRecords(n);
+        const { success, response, error } = await RecordsService.getAllDataRecords(
+            connection.connection_id,
+            connection.provider_config_key,
+            connection.environment_id,
+            model
+        );
+        expect(success).toBe(true);
+        expect(error).toBe(null);
+        expect(response?.records.length).toBe(n);
+        const timestampRegex = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{6}\+\d{2}:\d{2}$/;
+        expect(response?.records[0]?.['_nango_metadata']).toMatchObject({
+            first_seen_at: expect.stringMatching(timestampRegex),
+            last_modified_at: expect.stringMatching(timestampRegex),
+            last_action: 'ADDED',
+            deleted_at: null,
+            cursor: expect.stringMatching(/^[A-Za-z0-9+/]+={0,2}$/) // base64 encoded string
+        });
+        expect(response?.next_cursor).toBe(null); // no next page
+    });
```

Comment:
adding a test to make sure the format of metadata isn't modified with those changes